cli/main.rs

```diff
@@ -123,103 +124,177 @@ pub fn print_file_info(worker: &Worker, url: &str) {
   }
 }
 
-fn main() {
-  #[cfg(windows)]
-  ansi_term::enable_ansi_support().ok(); // For Windows 10
+fn get_worker_and_state(
+  flags: DenoFlags,
+  rest_argv: Vec<String>,
+) -> (Worker, ThreadSafeState) {
+  let state = ThreadSafeState::new(flags, rest_argv, ops::op_selector_std);
+  let worker = Worker::new(
+    "main".to_string(),
+    startup_data::deno_isolate_init(),
+    state.clone(),
+  );
 
-  log::set_logger(&LOGGER).unwrap();
-  let args = env::args().collect();
-  let (mut flags, mut rest_argv) =
-    flags::set_flags(args).unwrap_or_else(|err| {
-      eprintln!("{}", err);
-      std::process::exit(1)
-    });
+  (worker, state)
+}
 
-  log::set_max_level(if flags.log_debug {
-    LevelFilter::Debug
-  } else {
-    LevelFilter::Warn
-  });
+fn info_command(flags: DenoFlags, argv: Vec<String>) {
+  let (mut worker, state) = get_worker_and_state(flags, argv);
 
-  if flags.fmt {
-    rest_argv.insert(1, "https://deno.land/std/prettier/main.ts".to_string());
-    flags.allow_read = true;
-    flags.allow_write = true;
-  }
+  let main_module = state.main_module().unwrap();
+  let main_future = lazy(move || {
+    // Setup runtime.
+    js_check(worker.execute("denoMain()"));
+    debug!("main_module {}", main_module);
 
-  let should_prefetch = flags.prefetch || flags.info;
-  let should_display_info = flags.info;
+    let main_url = root_specifier_to_url(&main_module).unwrap();
 
-  let state = ThreadSafeState::new(flags, rest_argv, ops::op_selector_std);
-  let mut worker = Worker::new(
-    "main".to_string(),
-    startup_data::deno_isolate_init(),
-    state.clone(),
+    worker
+      .execute_mod_async(&main_url, true)
+      .and_then(move |worker| {
+        print_file_info(&worker, &main_module);
+        worker.then(|result| {
+          js_check(result);
+          Ok(())
+        })
+      }).map_err(|(err, _worker)| print_err_and_exit(err))
+  });
+  tokio_util::run(main_future);
+}
```

Comment:
@ry PTAL at it. It seem that `execute_mod_async` does not evaluate and run file if `prefetch` is set to true, so simple letting worker finish the job seems to work fine. I'm not sure though